keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -1107,114 +1120,133 @@ claims.
+	// parameters that will be used by the driver when allocating a
+	// resource that uses this class. A dynamic resource driver can
+	// distinguish between parameters stored here and and those stored in
+	// ResourceClaimSpec.
+	// +optional
+	ParametersRef *ResourceClassParametersReference
 
 	// Only nodes matching the selector will be considered by the scheduler
 	// when trying to find a Node that fits a Pod when that Pod uses
 	// a ResourceClaim that has not been allocated yet.
 	//
-	// Setting this field is optional. If nil, all nodes are candidates.
+	// Setting this field is optional. If null, all nodes are candidates.
+	// +optional
 	SuitableNodes *core.NodeSelector
 }
+```
+
+A copy of the driver name is necessary to enable usage
+of the claim by kubelet in case that the ResourceClass got
+removed in the meantime. It also helps the resource driver
+to determine whether it needs to handle a claim that got
+marked for deletion.
 
-// ResourceClaim is created by users to describe which resources they need.
+```
+// ResourceClaim describes which resources are needed by a resource consumer.
 // Its status tracks whether the resource has been allocated and what the
 // resulting attributes are.
+//
+// This is an alpha type and requires enabling the DynamicResourceAllocation
+// feature gate.
 type ResourceClaim struct {
 	metav1.TypeMeta
-
-	// The driver must set a finalizer here before it attempts to allocate
-	// the resource. It removes the finalizer again when a) the allocation
-	// attempt has definitely failed or b) when the allocated resource was
-	// deallocated. This helps to ensure that resources are not leaked
-	// during normal operation of the cluster.
-	//
-	// It cannot prevent force-deleting a ResourceClaim by clearing its
-	// finalizers (something that users should never do without being aware
-	// of the consequences) or help when the entire cluster gets deleted.
-	//
-	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// Standard object metadata
+	// +optional
 	metav1.ObjectMeta
 
 	// Spec describes the desired attributes of a resource that then needs
 	// to be allocated. It can only be set once when creating the
 	// ResourceClaim.
-	Spec core.ResourceClaimSpec
+	Spec ResourceClaimSpec
 
 	// Status describes whether the resource is available and with which
 	// attributes.
+	// +optional
 	Status ResourceClaimStatus
 }
+```
+
+The driver must set a finalizer in a ResourceClaim before it attempts to allocate
+the resource. It removes the finalizer again when a) the allocation
+attempt has definitely failed or b) when the allocated resource was
+deallocated. This helps to ensure that resources are not leaked
+during normal operation of the cluster.
+
+It cannot prevent force-deleting a ResourceClaim by clearing its
+finalizers (something that users should never do without being aware
+of the consequences) or help when the entire cluster gets deleted.
```

Comment:
What do you mean by "or help" here? Do you mean it also can't prevent the Claims from being deleted with then entire cluster gets deleted? Is that necessary to mention?

Reply:
It's one of two situations that a resource driver author has to consider where they cannot rely on Kubernetes to trigger the resource deallocation. I think it is worth mentioning because it might get overlooked, but I don't feel strongly about this.

Reply:
I don't feel very strongly either, I think it's fine.